src/raft/raft.rs

```diff
@@ -792,11 +830,89 @@ impl<T: Storage> Raft<T> {
                 return Ok(());
             }
         }
-        match self.state {
-            StateRole::Candidate => self.step_candidate(m),
-            StateRole::Follower => self.step_follower(m),
-            StateRole::Leader => self.step_leader(m),
+
+
+        match m.get_msg_type() {
+            MessageType::MsgHup => {
+                if self.state != StateRole::Leader {
+                    let ents = self.raft_log
+                        .slice(self.raft_log.applied + 1,
+                               self.raft_log.committed + 1,
+                               raft_log::NO_LIMIT)
+                        .unwrap();
```

Comment:
What I mean is to keep the [original expect](https://github.com/pingcap/tikv/blob/master/src/raft/raft.rs#L704) in our code.